### ISSUE TITLE
[Feature request]: migrate to react-codemirror

### ISSUE THREAD

**Issue by bpoulaindev** [usebruno/bruno#1212]
Migrate to [react-codemirror](https://github.com/uiwjs/react-codemirror) instead of codemirror
This would provide a react component, better suited for a react app instead of building the component through a function
The current component is **300+ lines long** and is getting harder and harder to maintain
Linked issue : #1029

**Comment by Its-treason** [comment]
Sounds like a good idea, this would also upgrade code-mirror to Version 6.

There are also some weird workarounds for code-mirror and Next.js Server-Side-Rendering: https://github.com/usebruno/bruno/blob/8ddec6bf0cf04be44ecf6d0b1777cf2c246e64e6/packages/bruno-app/src/pages/_app.js#L19-L31

https://github.com/usebruno/bruno/blob/8ddec6bf0cf04be44ecf6d0b1777cf2c246e64e6/packages/bruno-app/src/pages/Bruno/index.js#L13-L44

Maybe we can also think about moving from Next.js to Vite or something like this, because we are not using any Next.js features.

**Comment by bpoulaindev** [comment]
You're right @Its-treason , Next JS is better suited for SSR and SEO optimization, Vite would be lighter and faster, hence its name

**Comment by bpoulaindev** [comment]
I know the Typescript debate isn't settled yet, but i'de like to point out, this component would really benefit from having explicit types : 
<img width="720" alt="image" src="https://github.com/usebruno/bruno/assets/64689165/04b66084-8079-4658-a4b7-0c1a0e8c293c">

**Comment by bpoulaindev** [comment]
Trying out something, the following basic code can already provide the following render : 

<img width="717" alt="image" src="https://github.com/usebruno/bruno/assets/64689165/7d85df23-d5e3-47d4-8db3-890374cf9317">

<img width="628" alt="image" src="https://github.com/usebruno/bruno/assets/64689165/daf5104d-e04e-4f77-a498-da7649998c79">

**Comment by bpoulaindev** [comment]
Progress has been made : 
<img width="520" alt="image" src="https://github.com/usebruno/bruno/assets/64689165/a1db3450-4d3b-4132-86f1-3e826de15269">

Could use a bit more styling in the search component, about 20 themes are available, this isn't the definitive one for dark mode. We could also include, in the future, the possibility for users to switch between all of those 20 mods.

**Comment by dan1xk** [comment]
Is there a possibility for this feature to resolve this? #1347

**Comment by bpoulaindev** [comment]
@dan1xk Absolutely, i ended up using Monaco Editor, here's the PR : [Monaco Editor](https://github.com/usebruno/bruno/pull/1450)
